cmd/osv-scanner/fix/noninteractive.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+func autoInPlace(ctx context.Context, r reporter.Reporter, opts osvFixOptions, maxUpgrades int) error {
+	r.Infof("Scanning %s...\n", opts.Lockfile)
+	f, err := lockfile.OpenLocalDepFile(opts.Lockfile)
+	if err != nil {
+		return err
+	}
+
+	g, err := opts.LockfileRW.Read(f)
+	f.Close()
+	if err != nil {
+		return err
+	}
+
+	res, err := remediation.ComputeInPlacePatches(ctx, opts.Client, g, opts.RemediationOptions)
+	if err != nil {
+		return err
+	}
+
+	patches, nFixed, nRemain := autoChooseInPlacePatches(res, maxUpgrades)
+	totalVulns := nFixed + nRemain + len(res.Unfixable)
+
+	r.Infof("Found %d vulnerabilities matching the filter\n", totalVulns)
+	r.Infof("Can fix %d/%d matching vulnerabilities by changing %d dependencies\n", nFixed, totalVulns, len(patches))
+
+	for _, p := range patches {
+		r.Infof("UPGRADED-PACKAGE: %s,%s,%s\n", p.Pkg.Name, p.OrigVersion, p.NewVersion)
+	}
+	r.Infof("REMAINING-VULNS: %d\n", totalVulns-nFixed)
+	r.Infof("UNFIXABLE-VULNS: %d\n", len(res.Unfixable))
+
+	r.Infof("Rewriting %s...\n", opts.Lockfile)
+
+	return lf.Overwrite(opts.LockfileRW, opts.Lockfile, patches)
+}
+
+// returns the top {maxUpgrades} compatible patches, the number of vulns fixed, and the number of potentially fixable vulns left unfixed
+// if maxUpgrades is < 0, do as many patches as possible
+func autoChooseInPlacePatches(res remediation.InPlaceResult, maxUpgrades int) ([]lf.DependencyPatch, int, int) {
+	seenVKs := make(map[resolve.VersionKey]struct{})
+	type vulnKey struct {
+		id string
+		vk resolve.VersionKey
+	}
+	uniqueVulns := make(map[vulnKey]struct{})
+	var patches []lf.DependencyPatch
+	numFixed := 0
+
+	for _, p := range res.Patches {
+		vk := resolve.VersionKey{
+			PackageKey: p.Pkg,
+			Version:    p.OrigVersion,
+		}
+
+		if _, seen := seenVKs[vk]; maxUpgrades != 0 && !seen {
+			seenVKs[vk] = struct{}{}
+			patches = append(patches, p.DependencyPatch)
+			maxUpgrades--
+			numFixed += len(p.ResolvedVulns)
+		}
+
+		for _, rv := range p.ResolvedVulns {
+			uniqueVulns[vulnKey{id: rv.Vulnerability.ID, vk: vk}] = struct{}{}
+		}
+	}
+
+	return patches, numFixed, len(uniqueVulns) - numFixed
+}
+
+func autoRelock(ctx context.Context, r reporter.Reporter, opts osvFixOptions, maxUpgrades int) error {
+	r.Infof("Resolving %s...\n", opts.Manifest)
+	f, err := lockfile.OpenLocalDepFile(opts.Manifest)
+	if err != nil {
+		return err
+	}
+
+	manif, err := opts.ManifestRW.Read(f)
+	f.Close()
+	if err != nil {
+		return err
+	}
+
+	opts.Client.PreFetch(ctx, manif.Requirements, manif.FilePath)
+	res, err := resolution.Resolve(ctx, opts.Client, manif)
+	if err != nil {
+		return err
+	}
+
+	if errs := res.Errors(); len(errs) > 0 {
+		r.Warnf("WARNING: encountered %d errors during dependency resolution:\n", len(errs))
+		r.Warnf(resolutionErrorString(res, errs))
+	}
+
+	res.FilterVulns(opts.MatchVuln)
+	// TODO: count vulnerabilities per unique version as scan action does
```

Comment:
It's probably not too much work for just the counts, as long as we still take the all-or-nothing approach to resolving the vulns.
I think I'll look at this after I port the interactive mode, since it might change how I list the vulnerabilities there.